Blish HUD/Controls/_Types/UndoRedoStack.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ *  This code is heavily adapted from the Myra UndoRedoStack (https://github.com/rds1983/Myra/blob/a9dbf7a1ceedc19f9e416c754eaf38e89a89a746/src/Myra/Graphics2D/UI/TextEdit/UndoRedoStack.cs)
+ *
+ *  MIT License
+ *
+ *  Copyright (c) 2017-2020 The Myra Team
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+using System.Collections.Generic;
+
+namespace Blish_HUD.Controls {
+    internal class UndoRedoStack {
+
+        private readonly Stack<UndoRedoRecord> _stack = new Stack<UndoRedoRecord>();
+
+        public void Reset() {
+            _stack.Clear();
+        }
+
+        public UndoRedoRecord Pop() {
```

Comment:
Idk, if you should return an `UndoRedoRecord`-Object bc you are also not using those as inputs in the `Make`-Methods. Feels a little bit inconsistent that way. But don't ask me of a better way (bc the consumer needs to know the OperationType at least). Can also stay at is, just wanted to note.

Reply:
That's fair.  I don't expect this portion of the work to get a lot of "traffic" in terms of future changes, so I probably will leave it as is now even if only to simply the implementation on the TextInputBase side of things.

Reply:
Can be resolved then ^-^

Reply:
Thanks!